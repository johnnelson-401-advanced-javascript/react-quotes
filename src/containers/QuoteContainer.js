import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getQuote, getQuotesLoading } from '../selectors/quote-selectors';
import { setQuote } from '../actions/quoteActions';
import Quote from '../components/Quotes/Quote';

export default function OneQuote() {
  const quote = useSelector(state => getQuote(state));
  const dispatch = useDispatch();
  const loading = useSelector(state => getQuotesLoading(state));

  useEffect(() => {
    if(!quote) {
      dispatch(setQuote());
    }
  }, []);
  
  if(loading) return <h1>Loading . . . </h1>;

  function handleClick() {
    dispatch(setQuote());
  }

  return (
    <>
      <Quote quote={quote} handleClick={handleClick} />
    </>
  );
}
