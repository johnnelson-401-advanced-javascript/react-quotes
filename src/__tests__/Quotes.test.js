import React from 'react';
import { mount } from 'enzyme';
import QuoteContainer from '../containers/QuoteContainer';

describe('Quote Container', () => {
  it('Renders a Quote Container', () => {
    const quote = { image: 'string.com', character: 'Name', quote: 'Test' };
    const handleClick = jest.fn();
    const wrapper = mount(<QuoteContainer quote={quote} handleClick={handleClick}/>);
    expect(wrapper).toMatchSnapshot();
  });
  
});
