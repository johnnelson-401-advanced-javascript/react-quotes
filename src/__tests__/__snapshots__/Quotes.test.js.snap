// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes Renders a list of Quotes 1`] = `
<Fragment>
  <ul>
    <li
      key="Test"
    >
      <Quote
        character="Name"
        image="string.com"
        quote="Test"
      />
    </li>
  </ul>
</Fragment>
`;
